.snippets/code/tutorials/polkadot-sdk/parachains/zero-to-hero/build-custom-pallet/src/lib.rs

```diff
@@ -182,3 +182,146 @@ pub mod pallet {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate as custom_pallet;
+    use frame_support::{assert_noop, assert_ok, derive_impl, traits::ConstU32};
+    use sp_runtime::BuildStorage;
+
+    type Block = frame_system::mocking::MockBlock<Test>;
+
+    // Configure a mock runtime to test the pallet.
+    frame_support::construct_runtime!(
+        pub enum Test
+        {
+            System: frame_system,
+            CustomPallet: custom_pallet,
+        }
+    );
+
+    #[derive_impl(frame_system::config_preludes::TestDefaultConfig)]
+    impl frame_system::Config for Test {
+        type Block = Block;
+    }
+
+    impl Config for Test {
+        type RuntimeEvent = RuntimeEvent;
+        type CounterMaxValue = ConstU32<100>;
+    }
+
+    // Helper function to build genesis storage according to the mock runtime.
```

Comment:
```suggestion
    /// Helper function to build genesis storage according to the mock runtime.
```

Reply:
public functions should have triple function as it is then used as rust-docs, double slash is code comment.